File: Anya_Seth/Sliding Window/Count Number of Nice Subarrays.java
class Solution {
    public int numberOfSubarrays(int[] nums, int k){
        return niceCount(nums, k) - niceCount(nums, k-1);
    }
    public static int niceCount(int[] nums, int k){
        int start = 0;
        int end = 0;
        int niceCount = 0;
        while(end < nums.length){
            if(nums[end] % 2 != 0){
                k--;
            }
            while(k < 0){
            if(nums[start] % 2 != 0){
                k++;
            }
                start++;
            }
            niceCount += (end - start) + 1;
            end++;
        }
        return niceCount;
    }
}

File: NikhilKumarGupta/Slidingwindow/Count Number of Nice Subarrays.cpp
class Solution {
    public:
        int numberOfSubarrays(vector<int>& nums, int k) {
            int start=0;
            int count=0;
            int odd=0;
    
            for(int i=0;i<nums.size();i++){
                if(nums[i]==1){
                    odd++;
                }
    
                while(odd>k){
                    odd--;
                    start++;
                }
                if(odd==k){
                    count++;
                }
            }
            return count;
        }
    };